docs/arch/adr-0001-user-preferences-storage.md

```diff
@@ -0,0 +1,57 @@
+2. Voucher holders, who should be able to access and modify only their own stored preferences
+
+Options for access management to stored preferences considered here:
+
+1. [Auth0](#auth0) accounts with custom user metadata properties
+2. [Cognito](#cognito) accounts with S3 files
+3. []No authentication](#no-authentication) with S3 files
+
+
+
+### Auth0
+
+[Auth0](https://auth0.com) provides user authentication, authorization, and account management as a service. They offer unlimited free service to open-source, non-profit projects via their [Open Source Program](https://auth0.com/pricing). Their documentation includes [React samples](https://auth0.com/docs/quickstart/spa/react).
+
+The site preferences could be stored directly with the user via the [user metadata](https://auth0.com/docs/users/concepts/overview-user-metadata). (Cognito also supports custom user metadata.) If the user preferences are stored externally instead, however, accessing the resource with Auth0 would require using their now deprecated [delegation](https://auth0.com/docs/api-auth/intro#delegation) feature that may become unsupported at any time, requiring the project to switch to a different service or API feature.
+
+It is unclear how counselor access would be managed to the voucher holder accounts if the user preferences are stored as user metadata. Either the counselor would have to log in as the voucher holder, instead of having separate counselor accounts, or Auth0 access management would need to be modified somehow to allow counselor accounts to access and modify other accounts' metadata. Either option would involve defeating valuable aspects of account management and would likely prove difficult to implement.
+
+
+### Cognito
+
+Similar to Auth0, [Cognito](https://docs.aws.amazon.com/cognito/latest/developerguide/what-is-amazon-cognito.html) is the authentication, authorization, and user account management service provided by AWS. The service is free [up to 50,000 monthly active users](https://aws.amazon.com/cognito/pricing/), which should be plenty to cover this project; BHA has 12,000 ECHO vouchers, and serves 22,500 total, including other programs, according to [this](https://www.bostonhousing.org/en/For-Section-8-Leased-Housing/Voucher-Programs/Expanding-Choice-in-Housing-Opportunities-\(ECHO\).aspx).
+
+AWS has integrated Cognito with S3 bucket access controls, as described [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_s3_cognito-bucket.html). Also, there is a JavaScript library, [aws-amplify](https://github.com/aws-amplify/amplify-js), we could use for front-end integration. The library has Cognito authentication support and also a storage component that supports and defaults to using S3. The library appears to be popular and currently maintained. It supports both other authentication frameworks and storage backends, so should the project have need to switch either, it may be possible to do so with minimal code changes.
```

Comment:
The total extent of configuration required, and how it may be done (manual versus terraform) is unclear at this point. Given examples such as [this](https://github.com/terraform-providers/terraform-provider-aws) it looks like terraform should cover a significant portion of what is in the PGW screenshots @hectcastro linked to below.